client/rest/src/test/java/org/elasticsearch/client/RestClientIntegTests.java

```diff
@@ -205,6 +206,47 @@ public void testGetWithBody() throws IOException {
         bodyTest("GET");
     }
 
+    /**
+     * Ensure that basePath works as expected.
+     */
+    public void testPathPrefix() throws IOException {
+        // guarantee no other test setup collides with this one and lets it sneak through
+        final String uniqueContextSuffix = "/testPathPrefix";
+        final String pathPrefix = "base/" + randomAsciiOfLengthBetween(1, 5) + "/";
+        final int statusCode = randomStatusCode(getRandom());
+
+        HttpContext context = null;
+
+        try (final RestClient client =
+                RestClient.builder(new HttpHost(httpServer.getAddress().getHostString(), httpServer.getAddress().getPort()))
+                    .setPathPrefix(pathPrefix).build()) {
+
```

Comment:
can we test that the request fails before registering the specific context needed for this test?